internal/telemetry/telemetry.go

```diff
@@ -248,32 +248,35 @@ func (c *Client) defaultResource(ctx context.Context) (*resource.Resource, error
 	)
 }
 
-// randomId will read or generate a persistent `telemetry-random-id` file under `.cq` and return its value. If a new file is generated, c.newRandomId is set.
+// randomId will read or generate a persistent `telemetry-random-id` file and return its value.
+// First it will try reading ~/.cq/telemetry-random-id and use that value if found. If not, it will move on to ./cq/telemetry-random-id, first attempting a read and if not found, will create that file filling it with a newly generated ID.
+// If a directory with the same name is encountered, process is aborted and an empty string is returned.
+// If a new file is generated, c.newRandomId is set.
 func (c *Client) randomId() (string, error) {
```

Comment:
I really got confused with the functions the telemetry client has, it has NewRandomId, randomId, RandomId.. feels very confusing. 

Feels like the client should "get" id on creation, the id getting/generation should be functional code and tested rather than part of the client, the client should call that function to get the Id when its created. 

But maybe its me and I might have missed something, that is my 2 cents here ^^

Reply:
ok let's make a new `type randomId struct { fs afero.Whatever; newlyCreated bool }` and move methods there